File: src/main/java/ordilov/lissn/member/adapter/in/web/AuthController.java
package ordilov.lissn.member.adapter.in.web;

import lombok.RequiredArgsConstructor;
import ordilov.lissn.common.adapter.CommonResponse;
import ordilov.lissn.member.adapter.in.web.AuthDto.RefreshTokenDto;
import ordilov.lissn.member.application.port.in.AuthCommand;
import ordilov.lissn.member.domain.AuthInfo.RefreshInfo;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;


@RestController
@RequiredArgsConstructor
@RequestMapping("/auth")
public class AuthController {

  private final AuthCommand authCommand;

  @PostMapping("/refresh")
  public CommonResponse<RefreshInfo> refresh(@RequestBody RefreshTokenDto refreshTokenDto) {
    RefreshInfo refreshInfo = authCommand.refresh(refreshTokenDto.getRefreshToken());
    return CommonResponse.success(refreshInfo);
  }

}

File: src/main/java/ordilov/lissn/member/adapter/in/web/AuthController.kt
package ordilov.lissn.member.adapter.`in`.web

import ordilov.lissn.member.application.port.`in`.AuthCommand
import ordilov.lissn.member.domain.RefreshInfo
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class AuthController(
    val authCommand: AuthCommand
) {

    @PostMapping("/refresh")
    fun refresh(@RequestBody refreshTokenRequest: RefreshTokenRequest): ResponseEntity<RefreshInfo> {
        val refreshInfo = authCommand.refresh(refreshTokenRequest.refreshToken)
        return ResponseEntity.ok(refreshInfo)
    }
}